node/bft/src/primary.rs

```diff
@@ -576,24 +606,21 @@ impl<N: Network> Primary<N> {
         }
 
         // Retrieve the cached round and batch ID for this validator.
-        let signed_proposal = self.signed_proposals.read().get(&batch_author).copied();
-        if let Some((signed_round, timestamp, signed_batch_id, signature)) = signed_proposal {
-            // If the signed round is ahead of the peer's batch round, then then ignore the proposal.
+        if let Some((signed_round, signed_batch_id, signature)) =
+            self.signed_proposals.read().get(&batch_author).copied()
+        {
+            // If the signed round is ahead of the peer's batch round, then the validator is malicious.
             if signed_round > batch_header.round() {
-                bail!("Proposed a batch for a previous round ({})", batch_header.round());
+                // Proceed to disconnect the validator.
+                self.gateway.disconnect(peer_ip);
+                bail!("Malicious peer - proposed a batch for a previous round ({})", batch_header.round());
             }
-            // If the round matches and the batch ID differs, then check if the proposal is expired.
+
+            // If the round matches and the batch ID differs, then the validator is malicious.
             if signed_round == batch_header.round() && signed_batch_id != batch_header.batch_id() {
-                // Check if the proposal has expired.
-                match is_proposal_expired(now(), timestamp) {
-                    // If the proposal has expired, then remove the cached signature.
-                    true => self.signed_proposals.write().remove(&batch_author),
-                    // If the proposal has not expired, then disconnect the validator.
-                    false => {
-                        self.gateway.disconnect(peer_ip);
-                        bail!("Proposed another batch for the same round ({signed_round}) prior to expiration");
-                    }
-                };
+                // Proceed to disconnect the validator.
+                self.gateway.disconnect(peer_ip);
+                bail!("Malicious peer - proposed another batch for the same round ({signed_round})");
```

Comment:
I wouldn't call these 2 malicious peers. We've seen it happen on honest peers.

Reply:
The intention for this PR is to remove that behavior, but we can still be more lenient with the disconnects.

Reply:
Resolved with [7aaf276](https://github.com/AleoHQ/snarkOS/pull/3239/commits/7aaf2765bc7c53152f87b913c78b62892db54792)